ewatercycle/models/abstract.py

```diff
@@ -150,3 +168,20 @@ def time_step(self) -> float:
     def output_var_names(self) -> Iterable[str]:
         """List of a model's output variables."""
         return self.bmi.get_output_var_names()
+
+    def _check_parameter_set(self):
+        if not self.parameter_set:
+            # Nothing to check
+            return
+        model_name = self.__class__.__name__.lower()
+        if model_name != self.parameter_set.target_model:
+            raise ValueError(f'Parameter set has wrong target model, '
+                             f'expected {model_name} got {self.parameter_set.target_model}')
+        if self.parameter_set.supported_model_versions == set():
+            logger.warning(f'Model expects parameter set to support version {self.version}, '
+                           f'but parameter set supports any version')
+        elif self.version not in self.parameter_set.supported_model_versions:
+            raise ValueError(
+                f'Parameter set is not supported with version {self.version} of model, '
```

Comment:
`model = ewatercycle.models.Wflow('2020.1.13', parameters, forcing)` now returns an error about the parameter set not being compatible: `Parameter set is not supported with version 2020.1.13 of model, parameter set only supports {'2020.1.1'}`

I see two problems with this:

1. The first point of failure should be the model itself/the container. If the version is not in model.available_versions that should raise first. Only then should it proceed to check compatibility with parameterset.
2. I find the message confusing, since I'm not sure who is supposed to support what? Does the model support the parameterset? Perhaps it is clearer like this? 

```suggestion
                f'Parameter set is not compatible with version {self.version} of model, '
```

Reply:
Added check for version against available_version, which happens before pararmeter set checks.